bahmnicore-api/src/main/java/org/bahmni/module/bahmnicore/dao/impl/ObsDaoImpl.java

```diff
@@ -106,17 +119,19 @@
             query.append(" order by obs.obsDatetime desc ");
         }
 
-
         Query queryToGetObservations = sessionFactory.getCurrentSession().createQuery(query.toString());
-        queryToGetObservations.setMaxResults(limit);
+        queryToGetObservations.setMaxResults(limit != null ? limit : -1);
```

Comment:
what does -1 limit value means here ??

Reply:
Setting a negative number to hibernate `Query` will return all the reocrds